### ISSUE TITLE
Enhancement: name, status and departmental owner of LPARs on the IBM p-series AIX servers?

### ISSUE THREAD

**Issue by spezialist1** [samilliken/openDCIM#570]
Hi all!

I offer most safely are implemented method reading of the LPARs list from the IBM p-series servers with AIX OS.

Technically, use of the SNMP protocol for this purpose, as far as I know, is impossible (unlike the VMware ESXi servers). Besides, management even by the single IBM p-series server requires existence of the additional IBM hardware management console (HMC) server. IBM HMC - usual cheap Intel-based linux-server.

If to lower details to see the LPARs list (the virtual machines in IBM AIX terminology) and their status (is switched on or switched off), it is necessary to execute remotely on IBM HMC a command, for example:

root@opendcim-0# ssh viewuser@hmc_server lssyscfg -m ManagedServerName -r lpar -F name,state
Password:

AIX-test-restore,Not Activated
nim2-0,Running
vios_5_0,Running

where,
-- hmc_server - the IP-address or a name of the IBM HMC server. For simplicity, this value shall be entered into the "Primary IP/Host Name" field on the "Data Center Device Detail" page for the IBM p-series server interesting us;
-- viewuser - view only user who executes a remote input on the IBM HMC server (this user beforehand shall be created on IBM HMC). And, password entry can be avoided, using ssh keys/certificates. For this value it will be necessary to provide a custom user field in the new openDCIM version, for example, in openDCIM configuration;
-- ManagedServerName - a name (only the name is mandatory) the IBM p-series server interesting us on which we need to learn the LPARs list (virtual machines) and their status. Thus, this IBM p-series server shall be registered on the IBM HMC server which we use for request. The name ManagedServerName shall match precisely the name registered on the IBM HMC server and it needs to be entered into the "Label" field on the "Data Center Device Detail" page.

As a result of execution above the specified command we received, for example, the list from three LPARs (AIX-test-restore, nim2-0 and vios_5_0), and the virtual machine AIX-test-restore is now switched off, and two remaining LPARs are in a switched on status.

Besides, on the "Data Center Device Detail" page it is necessary to provide checkbox, by analogy with "ESX Server?" And also it will be necessary to create the script similar to poll_esx_inventory.php  for periodic updating of the LPARs list by crond.

I wait for your notes on this enhancement.
I am ready to be a tester in this case as I have direct access to the working IBM p-series servers.

Thanks for the support.

**Comment by spezialist1** [comment]
Hi all!

I suggest to implement in openDCIM above the specified enhancement again. And I am ready comprehensively to help to implement this decision and practically to test it in real DC.

Thanks for the support.

**Comment by samilliken** [comment]
Write the code and submit a pull request.  It will have to 100% come from
someone other than me or Wilbur as we do not have access to mainframes.

Scott

**Comment by spezialist1** [comment]
Hi Scott,

I and made, but unfortunately, I am not able to write professionally a PHP-code. Therefore decided to handle this sentence to you. It seems to me, from implementation of this function will benefit all community, and the openDCIM project will look even more interesting and more professional (and first of all advantage will be for the openDCIM developers ;-).

Very thanks for the support.

13.07.2016 04:02, Scott Milliken <notifications@github.com>
>Write the code and submit a pull request.  It will have to 100% come from
> someone other than me or Wilbur as we do not have access to mainframes.
> 
> Scott
> 
> On Tue, Jul 12, 2016 at 5:31 AM, spezialist1 <notifications@github.com>
> wrote:
> 
> > Hi all!
> >
> > I suggest to implement in openDCIM above the specified enhancement again.
> > And I am ready comprehensively to help to implement this decision and
> > practically to test it in real DC.
> >
> > Thanks for the support.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/samilliken/openDCIM/issues/570#issuecomment-231987367>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe/ABk8e65aZH6Fc8tu_WFEfYlAwNF6zMRrks5qU17zgaJpZM4EEPZj>
> > .
> >
> 
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/samilliken/openDCIM/issues/570#issuecomment-232226348

--
Чорнобиль форева!

-- реклама -----------------------------------------------------------
Поторопись зарегистрировать самый короткий почтовый адрес @i.ua
http://mail.i.ua/reg - и получи 1Gb для хранения писем

**Comment by Tops69** [comment]
This might help, this java application can pull all the information from the hardware management console which in turn can populate your database.
[HMC scanner](https://www.ibm.com/developerworks/community/wikis/home?lang=en#!/wiki/Power+Systems/page/HMC+Scanner)

**Comment by spezialist1** [comment]
Tops69, thanks for the link: I unfortunately didn't know about it. I will try to use in my work.

But after all there is a wish, that the openDCIM project could manipulate IBM LPAR's also as it is now carried out for VMware VM's.

Very thanks for the support.

**Comment by Jsphdcp31** [comment]
Thank you for your answer, i will try to test.
For information, my colleagues told me that the HMC v8 supports API restfull.

Thank you very Much

**Comment by spezialist1** [comment]
I agree with @Jsphdcp31, but I think that using classic ssh console access is a more universal method. Since mainframes live long and the people still use many more old HMC v7.
